src/processors/drive/muff_clipper/MuffClipper.cpp

```diff
@@ -0,0 +1,192 @@
+#include "MuffClipper.h"
+#include "../../ParameterHelpers.h"
+
+namespace
+{
+const auto cutoffRange = ParameterHelpers::createNormalisableRange (500.0f, 22000.0f, 1200.0f);
+float harmParamToCutoffHz (float harmParam)
+{
+    return cutoffRange.convertFrom0to1 (harmParam);
+}
+
+const auto sustainRange = ParameterHelpers::createNormalisableRange (0.4f, 2.0f, 1.0f);
+const auto levelRange = ParameterHelpers::createNormalisableRange (-60.0f, 0.0f, -9.0f);
+} // namespace
+
+MuffClipper::MuffClipper (UndoManager* um) : BaseProcessor ("Muff Clipper", createParameterLayout(), um)
+{
+    using namespace ParameterHelpers;
+    loadParameterPointer (sustainParam, vts, "sustain");
+    loadParameterPointer (harmParam, vts, "harmonics");
+    loadParameterPointer (levelParam, vts, "level");
+    clip1Param.setParameterHandle (getParameterPointer<chowdsp::FloatParameter*> (vts, "clip1"));
+    clip2Param.setParameterHandle (getParameterPointer<chowdsp::FloatParameter*> (vts, "clip2"));
+    smoothingParam.setParameterHandle (getParameterPointer<chowdsp::FloatParameter*> (vts, "smoothing"));
+    hiQParam = vts.getRawParameterValue ("high_q");
+
+    addPopupMenuParameter ("high_q");
+
+    uiOptions.backgroundColour = Colours::darkgrey.brighter (0.3f).withRotatedHue (0.2f);
+    uiOptions.powerColour = Colours::red.brighter (0.15f);
+    uiOptions.info.description = "Fuzz effect based on a single drive stage from the Electro-Harmonix Big Muff Pi.";
+    uiOptions.info.authors = StringArray { "Jatin Chowdhury", "Graham Breed" };
+}
+
+ParamLayout MuffClipper::createParameterLayout()
+{
+    using namespace ParameterHelpers;
+
+    auto params = createBaseParams();
+
+    createPercentParameter (params, "sustain", "Gain", 0.5f);
+    createPercentParameter (params, "harmonics", "Harm.", 0.65f);
+    createBipolarPercentParameter (params, "smoothing", "Smooth", 0.0f);
+    createBipolarPercentParameter (params, "clip1", "+Clip", 0.0f);
+    createBipolarPercentParameter (params, "clip2", "-Clip", 0.0f);
```

Comment:
From looking at the output in an oscilloscope I think these names might be flipped?